lib/config/defaults.js

```diff
@@ -1435,6 +1435,14 @@ const getResolveDefaults = ({ cache, context, targetProperties, mode }) => {
 		extensions: [...jsExtensions]
 	});
 
+	/** @type {function(): ResolveOptions} */
+	const styleDeps = () => ({
+		conditionNames: ["style"],
+		mainFields: ["style", "main", "..."],
```

Comment:
Don't think we need the `main` here, because it already in list, please try to remove it from here and test it